src/platform_impl/macos/window.rs

```diff
@@ -1151,6 +1151,20 @@ impl WindowExtMacOS for UnownedWindow {
                 .setHasShadow_(if has_shadow { YES } else { NO })
         }
     }
+
+    #[inline]
+    fn focus(&self) {
+        unsafe {
+            self.ns_window.makeKeyWindow();
```

Comment:
Like most things on `NSWindow`, this can probably only be run on the main thread, see the related `macos::util::async::make_key_and_order_front_async`

Reply:
Great, I will fix it according to your suggestion.